sdk/monitor/monitor-query/test/public/logsClient.spec.ts

```diff
@@ -0,0 +1,139 @@
+    // by coincidence one of the tests tries this same query.
+    await queryLogsWithTestRetries({
+      query: standardQuery,
+      maxTries: 20,
+      secondsBetweenQueries: 10
+    });
+  });
+
+  it("queryLogs", async () => {
+    const singleQueryLogsResult = await client.queryLogs(monitorWorkspaceId, standardQuery);
+
+    const tables: Table[] | undefined = singleQueryLogsResult.tables;
+
+    if (tables == null || tables.length === 0) {
+      throw new Error("No tables in result");
+    }
+
+    // TODO: the actual types aren't being deserialized (everything is coming back as 'string')
+    // this is incorrect, it'll be updated.
+
+    assert.equal(tables[0].name, "PrimaryResult");
+    assert.deepEqual(
+      tables[0].columns.map((c) => c.name),
+      ["Name", "Target", "TestRunId"]
+    );
+    assert.deepEqual(tables[0].rows?.[0], ["testSpan", "testSpan", testRunId.toString()]);
+  });
+
+  it("queryLogsBatch", async () => {
+    const batchRequest: BatchRequest = {
+      requests: [
+        // QueryBody
+        {
+          id: "arbitraryOperationName",
+          // RestError: The request had some invalid properties
+          workspace: monitorWorkspaceId,
+          body: {
+            workspaceIds: [monitorWorkspaceId],
+            query: standardQuery
+          }
+        }
+      ]
+    };
+
+    const batchQueryLogsResult = await client.queryLogsBatch(batchRequest);
+    const tables = batchQueryLogsResult.responses?.[0].body?.tables;
+
+    assert.deepEqual(
+      tables?.[0].columns.map((c) => c.name),
+      ["Name", "Target", "TestRunId"]
+    );
+    assert.deepEqual(tables?.[0].rows?.[0], ["testSpan", "testSpan", testRunId.toString()]);
+  });
+
+  it("serverTimeoutInSeconds", async () => {
+    // TODO: serverTimeoutInSeconds is passed. We need to have a deterministic way to test this, however.
+    const query = await client.queryLogs(monitorWorkspaceId, "AppEvents | limit 1", {
+      serverTimeoutInSeconds: 10
+    });
+
+    assert.ok(query);
+  });
+
+  it("includeQueryStatistics", async () => {
+    const query = await client.queryLogs(monitorWorkspaceId, "AppEvents | limit 1", {
+      includeQueryStatistics: true
+    });
+
+    // TODO: statistics are not currently modeled in the generated code.
+
+    // do a very basic check that the statistics were returnedassert.
+    assert.ok(query.statistics?.query?.executionTime);
+  });
+
+  async function queryLogsWithTestRetries(args: {
+    query: string;
+    secondsBetweenQueries: number;
+    maxTries: number;
+  }): Promise<QueryLogsResult> {
+    for (let i = 0; i < args.maxTries; ++i) {
+      const result = await client.queryLogs(monitorWorkspaceId, args.query, {
+        timespan: last30Minutes
+      });
+
+      const numRows = result.tables?.[0].rows?.length;
+
+      if (numRows != null && numRows > 0) {
+        return result;
+      }
+
+      await new Promise((resolve) => setTimeout(resolve, args.secondsBetweenQueries * 1000));
```

Comment:
Seems like these tests are going to be slow, and we shouldn't need the timeout when doing playback, right?

Reply:
Very true. I think there's a way for me to "not record" pieces so what I'll probably do is skip recording when doing these calls, and only turn it back on in each of the underlying steps.

The "setup" part is not critical for recording anyways.

Reply:
There is a way to fix this properly when I use recordings. I'm having "unrelated to this" issues so this first PR is going to go with recordings shut off.